### PR TITLE
De-dupe keywords server side

### LINKED ISSUE #2734: Problem depositing after item is saved as draft with duplicate keywords
See the video below. 

Before the video starts I had opened a version of this item and saved it as a draft with duplicate keywords, both chosen from the FAST vocabulary options.

Then in the video you'll see me do this:
1. Open the item to edit
2. Delete the second keyword -- note the other keyword still shows as red/invalid
3. Click the deposit button. I'm taken back up to the keywords. 
4. Click inside the existing keyword box and then click outside the box.
5. Click the deposit button. Now it moves me a tiny bit up the page. Every time I click deposit it moves up the page a tiny bit.

https://images.zenhubusercontent.com/272543337/b4d992bb-5588-4954-8389-1881e51b1d2c/screen_recording_2022_09_19_at_4_49_11_pm.mov

If I scrolled back up you would see there aren't any errors showing on the page any more. From here I was able to save it as a draft. When I edited it again, the second keyword was still gone and I was able to click deposit and it worked.

Ideally, once the second keyword is deleted the validations would understand that there is no longer a problem and I should be able to deposit the item.

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made? 🤔

Fixes #2734 - Alternative to #2743 after discussion in standup with Amy.  Instead of trying to do complex validation client and servers side, we will instead do auto de-duplication keywords on the server side (preferring keywords with URIs that come from auto-complete selection) and then remove keyword validation (server/client side).

Also removes an unused keywords component and it's related test.

## How was this change tested? 🤨

Localhost browser, and updated tests.

### CODE REVIEW

File: app/forms/work_form.rb [hunk 1]
```diff
@@ -6,7 +6,6 @@
 class WorkForm < DraftWorkForm
   validates :abstract, :access, :title, presence: true, allow_nil: false
   validates :keywords, length: { minimum: 1, message: 'Please add at least one keyword.' }
-  validate :keywords_unique
```
Comment (peetucket) [inline app/forms/work_form.rb:9]: remove server side keyword validation

File: app/javascript/controllers/keywords_controller.js [hunk 2]
```diff
@@ -17,42 +11,4 @@ export default class extends Controller {
...
-    this.checkForDuplicates() // clear out the invalid selection class if there are no duplicates anymore
-  }
-
-  hasDuplicateKeywords() {
-    // determine if the user has entered duplicate keywords
-    const nodes = document.querySelectorAll('.keyword-row:not([style*="display: none"]) input[type="text"]')
-    const keywords = Array.from(nodes, target => target.value)
-    // a set cannot have duplicates but an array can, so if the set and array are different in size, we have dupes
-    return (new Set(keywords)).size !== keywords.length
-  }
```
Comment (peetucket) [inline app/javascript/controllers/keywords_controller.js:57]: remove all of the client side javascript that was specific to keyword validation (there are generic methods for adding/removing associations that are used to add/remove keywords)

File: app/forms/draft_work_form.rb [hunk 3]
```diff
@@ -175,4 +176,18 @@ def model_name
   def will_assign_doi?
     (collection.doi_option == 'depositor-selects' && assign_doi) || collection.doi_option == 'yes'
   end
+
+  # de-dupe keywords, preferring to throwing away the free text entry duplicate(s) without a URI
+  def dedupe_keywords
+    return if work_version.keywords.size.zero?
+
+    grouped_keywords = Keyword.where(work_version:).group_by(&:label) # group keywords by the label
+    grouped_keywords.each do |_label, group|
```
Comment (justinlittman) [inline app/forms/draft_work_form.rb:185]: Slight suggestion: `grouped_keywords.values.each`

**Review by justinlittman — APPROVED** [review]

File: app/forms/draft_work_form.rb (hunk 3 above)
Comment (peetucket) [reply app/forms/draft_work_form.rb:185]: sounds good, in fact rubocop suggested `grouped_keywords.each_value` which I didn't know existed, and will use.

**Pull request merged**